src/routes/safe/components/Apps/ManageApps.js

```diff
@@ -79,15 +75,23 @@ const ManageApps = ({ appList, onAppAdded, onAppToggle }: Props) => {
 
   const cleanAppInfo = () => setAppInfo(APP_INFO)
 
-  const safeAppValidator = async (value) => {
-    const appInfo = await getAppInfoFromUrl(value)
-
-    if (appInfo.error) {
+  const customRequiredValidator = (value) => {
+    if (!value || !value.length) {
       setAppInfo(APP_INFO)
-      return 'This is not a valid Safe app.'
+      return 'Required'
     }
+  }
 
-    setAppInfo({ ...appInfo })
+  const getIpfsLinkFromEns = async (name) => {
```

Comment:
This doesn't have to be defined inside a component

Reply:
why do prefer this kind of functions outside the component, I mean, these functions do not change component state, but why is it that a bad practice?

Reply:
I'm not sure whether this is a good/bad practice, but here are my reasons why I don't like it:

1. When I think about component, for me it is about displaying things/view layer, and the low-level code for performing requests, validating things doesn't really belong there
2. Every time the component is rendered the variables are reassigned, there are possible performance drawbacks of doing that (Probably not very noticeable, but if we can avoid it by the cost as simple as moving function outside of the component, why not?). And just the logic - why make something reassign if it doesnt need to?
3. Everything is defined inside the component is a property of the component, it cannot be 
easily reused/imported by different parts of the app. For example, validators are things that could be possibly used by other parts of the app in the future.
4. Previously in the safe team we had an unwritten rule that any file in our app shouldn't be more than 250 lines of code. If you define everything in the component/single file it's hard to follow that rule.
5. It's harder to follow the code when the component grows, when you have a lot of functions defined the rendering logic is hard to follow because of such structure:
```
const x = // some function definition
useEffect() // component hook
const y = // some function definition
const z = // some function definition
useEffect() // component hook
```
Things are getting mixed up. Of course it's not a problem when the component is small but when it gets more complex like `ManageApps` it becomes a problem

Reply:
done!